src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/validator/ActionPlanSessionValidatorTest.kt

```diff
@@ -62,38 +44,36 @@ internal class ActionPlanSessionValidatorTest {
       var exception = assertThrows<ValidationError> {
         actionPlanSessionValidator.validateUpdateAppointment(updateAppointmentDTO)
       }
+      assertThat(exception.errors.size).isEqualTo(1)
+      assertThat(exception.errors.first().field).isEqualTo("appointmentDeliveryAddress")
       assertThat(exception.errors.first().error).isEqualTo(Code.CANNOT_BE_EMPTY)
-      updateAppointmentDTO = UpdateAppointmentDTO(appointmentTime = OffsetDateTime.now(), durationInMinutes = 1, appointmentDeliveryType = AppointmentDeliveryType.IN_PERSON_MEETING_OTHER, appointmentDeliveryAddress = listOf())
-      exception = assertThrows {
-        actionPlanSessionValidator.validateUpdateAppointment(updateAppointmentDTO)
-      }
-      assertThat(exception.errors.first().error).isEqualTo(Code.CANNOT_BE_EMPTY)
-    }
-
-    @Test
-    fun `too few address fields for non nps office appointment throws validation error`() {
-      var updateAppointmentDTO = UpdateAppointmentDTO(appointmentTime = OffsetDateTime.now(), durationInMinutes = 1, appointmentDeliveryType = AppointmentDeliveryType.IN_PERSON_MEETING_OTHER, appointmentDeliveryAddress = listOf("firstline", "town", "county", "A1 1AA"))
-      var exception = assertThrows<ValidationError> {
-        actionPlanSessionValidator.validateUpdateAppointment(updateAppointmentDTO)
-      }
-      assertThat(exception.errors.first().error).isEqualTo(Code.INVALID_LENGTH)
     }
 
     @Test
-    fun `too many address fields for non nps office appointment throws validation error`() {
-      var updateAppointmentDTO = UpdateAppointmentDTO(appointmentTime = OffsetDateTime.now(), durationInMinutes = 1, appointmentDeliveryType = AppointmentDeliveryType.IN_PERSON_MEETING_OTHER, appointmentDeliveryAddress = listOf("firstline", "secondLine", "thirdLine", "town", "county", "A1 1AA"))
+    fun `empty address fields for non nps office appointment throws validation error`() {
+      var updateAppointmentDTO = UpdateAppointmentDTO(appointmentTime = OffsetDateTime.now(), durationInMinutes = 1, appointmentDeliveryType = AppointmentDeliveryType.IN_PERSON_MEETING_OTHER, appointmentDeliveryAddress = AddressDTO("", "", "", "", ""))
       var exception = assertThrows<ValidationError> {
         actionPlanSessionValidator.validateUpdateAppointment(updateAppointmentDTO)
       }
-      assertThat(exception.errors.first().error).isEqualTo(Code.INVALID_LENGTH)
+      assertThat(exception.errors.size).isEqualTo(4)
+      assertThat(exception.errors.elementAt(0).field).isEqualTo("appointmentDeliveryAddress.firstAddressLine")
```

Comment:
_nit_ could possibly be expressed as `containsExactly` (which asserts order and cardinality) and using the equality of the data class
```kt
assertThat(exception.errors).containsExactly(
  FieldError("appointmentDeliveryAddress.townOrCity", Code.CANNOT_BE_EMPTY),
  FieldError("appointmentDeliveryAddress.county", Code.CANNOT_BE_EMPTY),
  // ...
)
```

Reply:
thanks, much neater